control/controlclient/direct.go

```diff
@@ -76,6 +77,8 @@ type Direct struct {
 	popBrowser             func(url string) // or nil
 	c2nHandler             http.Handler     // or nil
 
+	dialPlan *atomic.Pointer[tailcfg.ControlDialPlan] // can be nil
```

Comment:
this is a pointer to a pointer. (is that intentional? feels very odd) if so, which pointer can be nil? both?

Reply:
Yes, this is a pointer-to-pointer, and no, I'm not amazingly happy with this; suggestions appreciated. This is because `atomic.Pointer` isn't copyable, and we need to thread the value through several layers of abstraction.

Either can be nil; if the outermost pointer is nil, it's because there's no dial plan storage provided (e.g. the zero value for `Options`). If the innermost (atomic) pointer is nil, it's because we haven't yet received a dial plan from the control server, for example on the first connection.

Reply:
how about `dialPlan func() *tailcfg.ControlDialPlan`?

Reply:
And then thread that through the entire stack with the `atomic.Pointer.Load` function as the top-level? Yeah, that should work; lemme try it.

Reply:
That cleaned things up a bit, but we still need to store a `ControlDialPlan` between instances of `Direct`, so this still needs to be a pointer-to-a-pointer. I could make a wrapper type for `atomic.Pointer` that just checks if the receiver is nil and returns nil, though? `syncs.NilableAtomicPointer[tailcfg.ControlDialPlan]` that does nothing if the receiver is nil on `Store` and returns `nil` on `Load`?

Reply:
Switched to an interface after chatting today; no more pointers-to-pointers!